specification/watcher/_types/Conditions.ts

```diff
@@ -17,44 +17,43 @@
  * under the License.
  */
 
-import { Dictionary } from '@spec_utils/Dictionary'
+import { Dictionary, SingleKeyDictionary } from '@spec_utils/Dictionary'
 import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { FieldValue } from '@_types/common'
+import { AdditionalProperty } from '@spec_utils/behaviors'
 
 export class AlwaysCondition {}
 
-export class ArrayCompareCondition {
-  array_path: string
-  comparison: string
-  path: string
+export class ArrayCompareOpParams {
   quantifier: Quantifier
-  value: UserDefinedValue
+  value: FieldValue
 }
 
-export class CompareCondition {
-  comparison?: string
-  path?: string
-  value?: UserDefinedValue
-  'ctx.payload.match'?: CompareContextPayloadCondition
-  'ctx.payload.value'?: CompareContextPayloadCondition
+export class ArrayCompareCondition
```

Comment:
Should this be a `@variants container` object instead of an `AdditionalProperty<enum, value>`? A container with a required property seems like it fits this case better?

Reply:
This is really a pair of (operator, parameters) where parameters do not depend on the operator kind. A container is more suited to variants where the structure members depend on the variant discriminant.

Reply:
Makes sense, thanks for the explanation!